File: sub-mysql-web-entity/src/main/java/org/github/mysql/web/base/entity/SysUserMbpEntity.java
package org.github.mysql.web.base.entity;

import java.time.LocalDateTime;
import com.baomidou.mybatisplus.annotation.*;
import org.github.base.IEntity;
import javax.annotation.Nullable;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * <p>
 * 系统用户
 * </p>
 *
 * @author JYD_XL
 * @since 2021-05-29
 */
@SuppressWarnings("all")
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_user")
@ApiModel(value="SysUserMbpEntity对象", description="系统用户")
public class SysUserMbpEntity extends IEntity {

    /** UID */
    private static final long serialVersionUID = 1L;

    @TableId(value = "user_id", type = IdType.AUTO)
    private @Nullable Long userId;

    /** 用户名 */
    @ApiModelProperty(value = "用户名")
    @TableField("username")
    private @Nullable String username;

    /** 密码 */
    @ApiModelProperty(value = "密码")
    @TableField("password")
    private @Nullable String password;

    /** salt */
    @ApiModelProperty(value = "salt")
    @TableField("salt")
    private @Nullable String salt;

    /** 状态  0：禁用 1：正常 */
    @ApiModelProperty(value = "状态  0：禁用 1：正常")
    @TableField("status")
    private @Nullable Integer status;

    /** 邮箱 */
    @ApiModelProperty(value = "邮箱")
    @TableField("email")
    private @Nullable String email;

    /** 手机号 */
    @ApiModelProperty(value = "手机号")
    @TableField("mobile")
    private @Nullable String mobile;

    /** 创建者ID */
    @ApiModelProperty(value = "创建者ID")
    @TableField("create_user_id")
    private @Nullable Long createUserId;

    /** 创建时间 */
    @ApiModelProperty(value = "创建时间")
    @TableField("create_time")
    private @Nullable LocalDateTime createTime;

    /** 0：未删除 1：已删除 */
    @ApiModelProperty(value = "0：未删除 1：已删除")
    @TableField("deleted")
    private @Nullable Integer deleted;

}

File: sub-mysql-center-entity/src/main/java/org/github/mysql/center/base/entity/SysUserMbpEntity.kt
@file:Suppress("UnusedImport", "unused")

package org.github.mysql.center.base.entity

import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableName
import java.io.Serializable
import java.time.LocalDateTime
import org.github.core.base.Entity

/**
 * 用户
 *
 * @author JYD_XL
 */
@TableName("t_sys_user")
open class SysUserMbpEntity : Entity() {

    /** 主键 */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    open var id: Long? = null

    /** 登录用户名 */
    @TableField("user_name")
    open var userName: String? = null

    /** 登录密码 */
    @TableField("user_pwd")
    open var userPwd: String? = null

    /** 加密方式 */
    @TableField("secret_algorithm")
    open var secretAlgorithm: String? = null

    /** 秘钥 */
    @TableField("secret_key")
    open var secretKey: ByteArray? = null

    /** 昵称 */
    @TableField("nice_name")
    open var niceName: String? = null

    /** 创建人 */
    @TableField("creator_name")
    open var creatorName: String? = null

    /** 创建时间 */
    @TableField("created_time")
    open var createdTime: LocalDateTime? = null

    /** 修改人 */
    @TableField("updater_name")
    open var updaterName: String? = null

    /** 修改时间 */
    @TableField("updated_time")
    open var updatedTime: LocalDateTime? = null

    companion object {

        private const val serialVersionUID = 1L

        const val ID : String = "id"

        const val USER_NAME : String = "user_name"

        const val USER_PWD : String = "user_pwd"

        const val SECRET_ALGORITHM : String = "secret_algorithm"

        const val SECRET_KEY : String = "secret_key"

        const val NICE_NAME : String = "nice_name"

        const val CREATOR_NAME : String = "creator_name"

        const val CREATED_TIME : String = "created_time"

        const val UPDATER_NAME : String = "updater_name"

        const val UPDATED_TIME : String = "updated_time"

    }

    override fun pkVal(): Serializable? {
        return id
    }

    override fun toString(): String {
        return get()
    }

}
